pkg/cmd/auth/setupgit/setupgit_test.go

```diff
@@ -22,7 +26,56 @@ func (gf *mockGitConfigurer) Setup(hostname, username, authToken string) error {
 	gf.hosts = append(gf.hosts, hostname)
 	return gf.setupErr
 }
+func TestNewCmdSetupGit(t *testing.T) {
+	tests := []struct {
+		name     string
+		cli      string
+		wantsErr bool
+		errMsg   string
+	}{
+		{
+			name:     "--force without hostname",
+			cli:      "--force",
+			wantsErr: true,
+			errMsg:   "cannot use `--force` without `--hostname`",
+		},
+		{
+			name:     "no error when --force used with hostname",
+			cli:      "--force --hostname ghe.io",
+			wantsErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			f := &cmdutil.Factory{}
+
+			argv, err := shlex.Split(tt.cli)
+			assert.NoError(t, err)
+
+			cmd := NewCmdSetupGit(f, func(opts *SetupGitOptions) error {
+				return nil
+			})
+
+			// TODO cobra hack-around
+			cmd.Flags().BoolP("help", "x", false, "")
```

Comment:
I was getting below error for the test cases to check flag.
**panic: unable to redefine 'h' shorthand in "setup-git" flagset: it's already used for "hostname" flag [recovered]**

I observed most of the test cases in gh cli used below code to get rid of this error. 
`cmd.Flags().BoolP("help", "x", false, "")`

So used it for setup-git's test-cases.

Reply:
Yeh it's super janky. The issue is that `-h` is used for hostname but it's configured outside each individual command, so in the tests we update help to have shorthand of `-x`. 🤮